src/arco_era5/update_model_level_native_vertical_zarr.py

```diff
@@ -0,0 +1,274 @@
+    coordinates to float32.
+    """
+
+    # It's possible to have coordinate metadata for coordinates which aren't
+    # actually used as dimensions of any variables (called 'non-index'
+    # coordinates), and some of the source NetCDF files use these, e.g. a scalar
+    # 'height' coordinate (= 2.0) in the NetCDF files for the 2-meter temperature
+    # variable tas. We remove these, for simplicity and since once the variables
+    # are combined in a single zarr dataset it won't be clear what these
+    # additional coordinates are referring to.
+    dataset = dataset.reset_coords(drop=True)
+
+    # Downcast lat and lon coordinates to float32. This is because there are
+    # small rounding-error (~1e-14 relative error) discrepancies between the
+    # float64 latitude coordinates across different source NetCDF files, and
+    # xarray_beam complains about this. After downcasting to float32 the
+    # differences go away, and the extra float64 precision isn't important to us.
+    # (Ideally you'd be able to specify a tolerance for the comparison, but this
+    # works for now.)
+    dataset = dataset.assign_coords(
+        latitude=dataset["latitude"].astype(np.float32),
+        longitude=dataset["longitude"].astype(np.float32)
+    )
+    return dataset
+
+def offset_along_time_axis(start_date: str, year: int, month: int, day: int, hour: int) -> int:
+    """Calculate offset in indices along the time axis, relative to start of the dataset.
+    # Note the length of years can vary due to leap years, so the chunk lengths
+    # will not always be the same, and we need to do a proper date calculation
+    # not just multiply by 365*24.
+    Args:
+        start_date (str): The start date in ISO format (YYYY-MM-DD).
+        year (int): The year.
+        month (int): The month.
+        day (int): The day.
+        hour (int): The hour.
+
+    Returns:
+        int: The offset in indices along the time axis.
+    """
+    time_delta = pd.Timestamp(year=year, month=month, day=day, hour=hour) - pd.Timestamp(start_date)
+    return int(time_delta.total_seconds() // 60 // 60)
+
+@dataclass
+class UpdateSlice(beam.PTransform):
+    """A Beam PTransform to write Zarr arrays from the xarray datasets and time offset."""
+
+    target: str
+    init_date: str
+
+    def apply(self, key: xb.Key, ds: xr.Dataset) -> None:
+        """Write Zarr arrays from the xarray datasets and time offset.
+
+        Args:
+            key (xb.Key): Offset dictionary for dimensions.
+            ds (xr.Dataset): Merged dataset for a single day.
+        """
+        logger.info("Inside the UpdateSlice function of the AR data.")
+        offset = key.offsets['time']
+        date = (datetime.datetime.strptime(self.init_date, '%Y-%m-%d') + 
+                datetime.timedelta(hours=offset))
+        zf = zarr.open(self.target)
+        region = slice(offset, offset + 1)
+        start_date = date.strftime('%Y-%m-%dT%H')
```

Comment:
We can get rid of the start and end date. We can display `ds.time.values()` directly.